src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -138,25 +137,28 @@ private void Append(string value)
             }
             else
             {
-                byte[] bytes = Encoding.UTF8.GetBytes(value);
-                DataFrameBuffer<byte> mutableDataBuffer = _dataBuffers[_dataBuffers.Count - 1] as DataFrameBuffer<byte>;
+                DataFrameBuffer<byte> mutableDataBuffer = (DataFrameBuffer<byte>)_dataBuffers[_dataBuffers.Count - 1];
                 if (mutableDataBuffer.Length == ReadOnlyDataFrameBuffer<byte>.MaxCapacity)
                 {
                     mutableDataBuffer = new DataFrameBuffer<byte>();
                     _dataBuffers.Add(mutableDataBuffer);
                     _nullBitMapBuffers.Add(new DataFrameBuffer<byte>());
-                    var offsetBuffer = new DataFrameBuffer<int>();
-                    _offsetsBuffers.Add(offsetBuffer);
-                    offsetBuffer.Append(0);
+                    mutableOffsetsBuffer = new DataFrameBuffer<int>();
+                    _offsetsBuffers.Add(mutableOffsetsBuffer);
+                    mutableOffsetsBuffer.Append(0);
                 }
-                mutableDataBuffer.EnsureCapacity(bytes.Length);
-                bytes.AsMemory().CopyTo(mutableDataBuffer.Memory.Slice(mutableDataBuffer.Length));
-                mutableDataBuffer.Length += bytes.Length;
-                mutableOffsetsBuffer.Append(mutableOffsetsBuffer[mutableOffsetsBuffer.Length - 1] + bytes.Length);
+                mutableDataBuffer.EnsureCapacity(value.Length);
+                value.CopyTo(mutableDataBuffer.Span.Slice(mutableDataBuffer.Length));
+                mutableDataBuffer.Length += value.Length;
+                mutableOffsetsBuffer.Append(mutableOffsetsBuffer[mutableOffsetsBuffer.Length - 1] + value.Length);
             }
             SetValidityBit(Length - 1, value == null ? true : false);
+
         }
 
+        // This is an immutable column, however this method exists to support Clone(). Keep this method private
+        private void Append(string value) => Append(value == null ? null : Encoding.UTF8.GetBytes(value));
```

Comment:
I'm not 100% sure, but is this only called with `null`? In general I don't think we should be using this method. Can you remove it somehow, to ensure no one calls it in the future, since it will cause needless memory consumption?

(note this can be done in a follow up PR)

Reply:
Removed